Read Text file:

In [4]:
import os
os.chdir('../data')
text = open('twentytens.txt').read()

FileNotFoundError: [Errno 2] No such file or directory: '../data'

Functions:

In [29]:
from collections import Counter
import os
import string
import json
import matplotlib.pyplot as plt
from IPython.display import Image

In [30]:
import os
os.chdir('../data_analysis')
%run functions.ipynb

### 2010s corpus set up:

In [31]:
text.index('200. Europe issues 2nd recall of drugs made by Ben Venue')

12975

In [32]:
no_toc_text=text[12975:]

In [33]:
doc_list=no_toc_text.split('End of Document')
len(doc_list)

201

In [8]:
token_docs=[]
for doc in doc_list:
    doc_tokens=tokenize(doc,lowercase=True,strip_chars='!,."')
    token_docs.append(doc_tokens)

In [9]:
flat_list=[]
for sublist in token_docs:
    flat_list.extend(sublist)

len(flat_list)

81184

Filter out stop words:

In [10]:
from nltk.corpus import stopwords
stoplist=stopwords.raw('english').split('\n')

In [11]:
filtered_flat_list=[]
for words in flat_list:
    if words not in stoplist:
        filtered_flat_list.append(words)

## Frequency Lists:

Top 100 words:

In [12]:
counted_toks=Counter(filtered_flat_list)
print(counted_toks.most_common(100))

[('drug', 903), ('recall', 759), ('said', 670), ('fda', 556), ('drugs', 516), ('us', 424), ('products', 401), ('recalled', 396), ('2019', 344), ('patients', 340), ('health', 308), ('company', 301), ('valsartan', 249), ('may', 228), ('blood', 225), ('body', 211), ('copyright', 206), ('product', 205), ('found', 204), ('recalls', 204), ('rights', 200), ('length:', 200), ('words', 200), ('load-date:', 200), ('tablets', 200), ('reserved', 199), ('also', 181), ('food', 179), ('medicines', 174), ('used', 170), ('lots', 170), ('pressure', 166), ('manufacturing', 162), ('pharmaceuticals', 159), ('ndma', 155), ('ranitidine', 152), ('administration', 151), ('cancer', 151), ('losartan', 150), ('market', 145), ('risk', 143), ('last', 141), ('pharma', 138), ('recalling', 138), ('news', 136), ('impurity', 136), ('agency', 135), ('inc', 128), ('adverse', 119), ('due', 118), ('manufactured', 116), ('levels', 114), ('cause', 113), ('pharmaceutical', 111), ('byline:', 109), ('zantac', 108), ('taking', 10

Top 50 bigrams

In [13]:
bigrams=get_ngram_tokens(flat_list,n=2)
count_bigrams=Counter(bigrams)
print(count_bigrams.most_common(50))

[('of the', 428), ('in the', 365), ('the fda', 349), ('the us', 291), ('to the', 254), ('the recall', 205), ('all rights', 200), ('rights reserved', 199), ('the company', 189), ('recall of', 170), ('the drug', 167), ('blood pressure', 165), ('from the', 155), ('have been', 151), ('food and', 141), ('and drug', 140), ('drug administration', 139), ('in a', 135), ('lots of', 130), ('by the', 127), ('on the', 121), ('due to', 115), ('that the', 113), ('for the', 110), ('words byline:', 109), ('copyright 2019', 104), ('said the', 103), ('according to', 102), ('with the', 99), ('reserved length:', 98), ('us food', 95), ('as a', 91), ('it is', 89), ('reserved section:', 88), ('of a', 87), ('to be', 84), ('to a', 82), ('has been', 80), ('used to', 77), ('is a', 77), ('said it', 76), ('the agency', 74), ('said in', 72), ('of its', 72), ('the drugs', 71), ('at the', 69), ('and the', 69), ('voluntary recall', 67), ('the product', 66), ('found in', 66)]


Top 50 trigrams

In [14]:
trigrams=get_ngram_tokens(flat_list,n=3)
count_tri=Counter(trigrams)
print(count_tri.most_common(50))

[('all rights reserved', 199), ('food and drug', 140), ('and drug administration', 134), ('rights reserved length:', 98), ('us food and', 93), ('rights reserved section:', 88), ('the us food', 85), ('in the us', 81), ('of the drug', 58), ('the us market', 54), ('used to treat', 52), ('adverse health consequences', 52), ('high blood pressure', 52), ('according to the', 52), ('from the us', 47), ('the fda said', 46), ('in a statement', 45), ('inc all rights', 44), ('the presence of', 43), ('voluntary recall of', 41), ('said in a', 41), ('the company said', 40), ('trace amounts of', 39), ('blood pressure and', 38), ('of blood pressure', 36), ('the united states', 33), ('drug administration (fda)', 31), ('use of or', 31), ('of or exposure', 31), ('or exposure to', 31), ('lots of losartan', 31), ('losartan potassium tablets', 29), ('global data point', 29), ('ltd all rights', 29), ('of losartan potassium', 29), ('have been recalled', 29), ('are being recalled', 28), ('the food and', 28), ('

Frequency List of Top 100 Words 
* "Valsartan" (249), "losartan" (150), and "ranitidine" (152) highlight specific drugs central to major controversies and recalls
* "Patients" (340), "health" (308), and "risk" (143) reflect a strong focus on public health and safety concerns
* "NDMA" (155) and "impurity" (136) point to increased scrutiny on manufacturing processes and contamination issues
* "Manufacturing" (162), "manufactured" (116), and "quality" (86) show a growing emphasis on production standards and failures
* "Food" (179), "administration" (151), and "agency" (135) show the FDA's important role in regulating and addressing crises

Top 50 Bigrams 
* "Blood pressure" and "lots of" point to specific health conditions and manufacturing issues central to product recalls
* "The company" and "said the" emphasize corporate statements and media framing of events
* "Due to," "have been," and "found in" shows contamination risk and increased transparency
* "The product" and "the drugs" reinforce the focus on specific pharmaceutical items and their impact

Top 50 Trigrams
* "All rights reserved" and related phrases reflect the formal, legal language typical of structured reporting
* "US food and" and "the US food" highlight the focus on regulatory efforts specific to the United States
* "Adverse health consequences" and "high blood pressure" link to key health risks associated with recalled drugs
* "The company said" and "said in a statement" reflect a heavy reliance on corporate communication and public statements
* "Voluntary recall of" and "are being recalled" show recalls being framed as proactive measures by companies
* "Trace amounts of" and "the presence of" show manufacturing flaws
* "Exposure to a" and "to a violative product" point to risks identified in products that failed regulatory standards


## Context Analyses (KWIC):

In [15]:
jj_words=['showed','safety','responsible','trust', 'trusted','forgiveness','consumer','example','rectification','sympathy']

In [19]:
words_10s=[]
for word in filtered_flat_list:
    if word in jj_words:
        words_10s.append(word)

len(words_10s)

152

In [21]:
count_word=Counter(words_10s)
count_word

Counter({'safety': 74,
         'showed': 15,
         'consumer': 40,
         'example': 10,
         'trust': 2,
         'responsible': 11})

### KWIC concordances:

#### 1. Safety

In [22]:
safety=make_kwic('safety',flat_list,win=6)
print_kwic(sort_kwic(safety))

                   last month they were reviewing the  safety  of ranitidine after online pharmacy valisure
                   last month they were reviewing the  safety  of ranitidine after online pharmacy valisure
      european medicines agency determines the drug’s  safety  the authorities said patients who regularly
                    week said they were reviewing the  safety  of ranitidineafter they found traces of
               sandra eskin project director for food  safety  at pew charitable trusts explained in
interpretation however undermines consumer health and  safety  and should be changed by legislation
                          updated in 2011 by the food  safety  modernization act which gave fda authority
                        the problem she said the food  safety  project director spent much of her
               the recall pending confirmation of the  safety  risk fda director general nela charade
               sandra eskin project director for food  safety  at p

#### 2. Example

In [23]:
example=make_kwic('example',flat_list,win=6)
print_kwic(sort_kwic(example))

         definition of a dietary supplement for  example  this would extend to a supplement
sexual enhancement and athletic performance for  example  in january an fda analysis of
         definition of a dietary supplement for  example  this would extend to a supplement
sexual enhancement and athletic performance for  example  in january an fda analysis of
              are accepted for cancer drugs for  example  but not for an allergy drug
                october and natpara is just one  example  of a growing list of drugs
             risk of developing the disease for  example  a daily dose of valsartan may
                     a good state of repair for  example  on september 8 2014 we observed
                      best to make the drug for  example  detroit-based caraco pharmaceutical laboratories issued four
                     a good state of repair for  example  on september 8 2014 we observed


#### 3. Showed

In [24]:
showed=make_kwic('showed',flat_list,win=6)
print_kwic(sort_kwic(showed))

                                                                               fda last week tests on animals  showed  that a particular substance ndma had
                                                                        of these drugs the contaminants first  showed  up in drugs that contained valsartan
                                                                        of these drugs the contaminants first  showed  up in drugs that contained valsartan
                                                                 branded and generic ranitidine products that  showed  some versions of the over-the-counter medicine
                                                                 branded and generic ranitidine products that  showed  some versions of the over-the-counter medicine
https://wwwfdagov/drugs/drug-safety-and-availability/laboratory-tests-ranitidine ) on ranitidine products and  showed  that some versions of the drug
https://wwwfdagov/drugs/drug-safety-and-availabili

#### 4. Consumer

In [25]:
consumer=make_kwic('consumer',flat_list,win=6)
print_kwic(sort_kwic(consumer))

                       for treating blood pressure to the  consumer  level this is an expansion of
        potassium/ hydrochlorothiazide tablets usp to the  consumer  level due to the presence of
                  health threats according to adam garber  consumer  watchdog at us pirgthe agency 'should
                 for over-the-counter products and at the  consumer  level for prescription products of all
                      it was earlier revealed that mcneil  consumer  health care was recalling 21 over
                          to a product recall that mcneil  consumer  health care originally announced on january
                    on january 15 was initiated following  consumer  complaints of a musty or mouldy
          violations including the failure to investigate  consumer  complaints and poor quality-control procedures that
                                      50 mg and 100 mg to  consumer  level in united states (us) as
                                      50 mg and 10

#### 5. Responsible

In [26]:
responsible=make_kwic('responsible',flat_list,win=6)
print_kwic(sort_kwic(responsible))

                      products for human use which is  responsible  for probing concerns over patient medicines
            into whether the contaminated medicine is  responsible  for any infant deaths although the
                   medical devices the agency also is  responsible  for the safety and security of
                      products for human use which is  responsible  for probing concerns over patient medicines
                   medical devices the agency also is  responsible  for the safety and security of
                      products for human use which is  responsible  for probing concerns over patient medicinesits
                      products for human use which is  responsible  for probing concerns over patient medicines
supplement manufacturers and distributors are legally  responsible  for marketing a safe product that
          potent these pharmaceuticals are cohen said  responsible  manufacturers and marketers of dietary supplements
                           

#### 6. Trust

In [27]:
trust=make_kwic('trust',flat_list,win=6)
print_kwic(sort_kwic(trust))

alexander smith our standards:the thomson reuters  trust  principles load-date: february 11 2019 affymax
                           all want to be able to  trust  that what we are dispensing is


### Observations:

KWIC Analysis of "Safety"
* Commonly mentioned in connection to drug recalls and reviews, highlighting its importance in product evaluations
* Frequently associated with phrases like "patient safety" and "food safety," showing a focus on protecting public health
* Appears in discussions of regulatory actions, such as "safety review" and "safety standards," underscoring the FDA’s role
* Often tied to specific health risks, including contamination and adverse effects, emphasizing the stakes for consumers
* Used in proactive contexts like "commitment to health and safety," showing companies’ efforts to rebuild trust
* Shows up in legislation contexts, such as "food safety modernization act," highlighting evolving regulatory frameworks
* Phrases like "no evidence of risk to safety" appear to reassure the public regarding product concerns
* Frequently appears with terms like "quality standards" and "effectiveness," tying safety to broader product assurances


KWIC Analysis of "Example"
* Often used to illustrate specific cases or scenarios, such as dietary supplements or cancer drugs
* Appears in contexts highlighting comparisons, like differences in drug risks or standards
* Provides context for explaining FDA actions or pharmaceutical issues, such as recalls or manufacturing lapses
* Helps clarify broader points with specific instances, like "a growing list of drugs" or observed conditions in facilities
* Frequently associated with health-related impacts or regulatory evaluations, making it a tool for better understanding key issues

KWIC Analysis of "Showed" 
* Primarily used to describe study or test results, emphasizing evidence-based findings in drug safety and product recalls
* Highlights contamination and safety risks, such as "some versions of the drug showed NDMA" or "growth of bacteria"
* Demonstrates the role of scientific studies in regulatory actions, with results like "the product contained sibutramine and phenolphthalein"
* Reflects large-scale health impacts with phrases like "nearly two million people were affected"
* Similar to earlier decades, "showed" continues to be associated with health risks and product evaluations, but the focus shifts toward more specific substances (e.g., valsartan, ranitidine) and broader population-level data
* Unlike earlier periods, there is an increasing emphasis on specific contaminants like NDMA and their widespread detection in various products
* Continues to underline the critical role of test results in shaping FDA decisions and public health advisories

KWIC Analysis of "Consumer" 
* Frequently associated with "consumer level," indicating recalls directly affecting consumers rather than higher supply chain levels
* Often linked to complaints or issues such as "consumer complaints of a musty or moldy smell," highlighting consumer feedback's role in triggering recalls
* Mentions of "consumer health" and "consumer safety" emphasize the direct impact of products on public well-being
* Connected to regulatory actions and new laws like "consumer protection law," showing evolving efforts to safeguard consumers
* Appears frequently in discussions of product recalls, particularly involving major brands like  McNeil Consumer Healthcare and Johnson & Johnson
* Highlights failures in quality control and the importance of addressing consumer concerns in maintaining trust
* Demonstrates how consumer watchdog groups and reports influence regulatory and corporate actions
* Reinforces the growing role of transparency and accountability in consumer-facing communications and recall decisions

KWIC Analysis of "Responsible"
* Frequently tied to regulatory oversight, such as the FDA being "responsible for the safety and security" of medicines and medical devices
* Used in contexts highlighting accountability, such as supplement manufacturers being "legally responsible for marketing a safe product"
* Reflects the role of industry stakeholders, with mentions of "responsible manufacturers and marketers" emphasizing corporate duty
* Associated with leadership and advocacy, such as the "CEO of the council for responsible nutrition," demonstrating a focus on public health and industry standards
* Often linked to investigations into contamination or patient safety concerns, underscoring the importance of identifying accountability in crises

KWIC Analysis of "Trust" 
* Appears in the context of credibility, with references to "trust principles," indicating the importance of maintaining journalistic or organizational standards
* Highlights consumer confidence, as seen in phrases like "trust that what we are dispensing," showing the critical need for reliable and safe products
* Demonstrates a focus on rebuilding or maintaining public trust, particularly in industries dealing with health and safety concerns


## Sentiment Analysis: VADER Scores

In [34]:
### Set up:

import os
import re
import csv
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

from nltk import tokenize
from nltk.sentiment.vader import SentimentIntensityAnalyzer

In [35]:
### Using VADER
sid = SentimentIntensityAnalyzer()

### text set up
tens_text=no_toc_text.lower().replace('\n','')

In [37]:
sid.polarity_scores(tens_text)

{'neg': 0.072, 'neu': 0.869, 'pos': 0.059, 'compound': -1.0}

### Observations: negative compound score, lowest positive sentiment score of all corpora